storage/badger/operation/common.go

```diff
@@ -76,7 +77,7 @@ func insert(key []byte, entity interface{}) func(*badger.Txn) error {
 		}
 
 		// persist the entity data into the DB
-		err = tx.Set(key, val)
+		err = terminateOnFullDisk(tx.Set(key, val))
```

Comment:
I don't think `tx.Set` actually does the write to disk. That's done by `tx.Commit`. It's worth testing this out to make sure

Reply:
Absolutely right! I did more digging and we have a dual-use problem:
- some parts of the code like [StoreLight in collection](https://github.com/onflow/flow-go/blob/710e4bf266d53882fa407374925ba23d9846655c/storage/badger/collections.go#L28-L35) actually use badger's inbuilt `Update` that has a similar flow to our own [tx wrapper's update](https://github.com/onflow/flow-go/blob/710e4bf266d53882fa407374925ba23d9846655c/storage/badger/transaction/tx.go#L26-L45) 
- so the question is: do we go back to the original model of applying this to `operation` and wrapping that call everywhere (not a fan of this personally) or do we make this disk crash logic only apply to `transaction.Update` calls?
- one approach is to do both, keep `terminateOnFullDisk ` in `operation` and wrap where `db.Update` is used, and wrap the `Commit()` in `transaction` with it as well

Reply:
Implemented the above approach without wrapping `db.Update` calls, minimum viable refactor